Public/Src/Utilities/KeyValueStore/RocksDb/RocksDbStore.cs

```diff
@@ -197,6 +197,15 @@ private struct DefaultOptions
                     .SetBlockBasedTableFactory(blockBasedTableOptions)
                     .SetPrefixExtractor(SliceTransform.CreateNoOp());
 
+                // Maximum number of information log files
+                m_defaults.DbOptions.SetKeepLogFileNum(1000);

```

Comment:
What performance implications does this have on other consumers of RocksDb in the codebase? Is it going to cause BuildXL to keep a zillion rocksDB log files?

Reply:
I'm afraid it would, yes. This is mainly useful in the case of lab builds; hence, one way to move forward is to add a flag defaulted to false, which we can enable for our use case. Another option is to recycle the log files, so we can just keep at most N log files with at most size S, and RocksDB would proceed in a ring fashion as it runs out of files.

Please let me know which option you think is best.

Reply:
I'd prefer the different flag approach. For the build engine keeping any more than one log file is going to be overkill